lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Addresses.hs

```diff
@@ -253,6 +257,83 @@ spec = describe "SHELLEY_ADDRESSES" $ do
                 (`shouldNotSatisfy` T.isPrefixOf "addr_test")
             ]
 
+    it "ADDRESS_LIST_06 - Change addresses are listed even after transaction is no longer pending" $ \ctx -> runResourceT $ do
+        let initPoolGap = 10
+
+        let verifyAddrs i p addr = do
+                liftIO (length (filter ((== Used) . (^. (#state . #getApiT))) addr)
+                    `shouldBe` p)
+                liftIO (length addr `shouldBe` i)
+
+        -- 1. create shelley wallet
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWalletWith ctx ("Wallet", "cardano-wallet", initPoolGap)
+
+        -- 2. list addresses of that wallet (for later comparison)
+        listAddresses @n ctx wSrc >>= verifyAddrs 50 10
+
+        -- 3. send a transaction from that wallet
+        addrs <- listAddresses @n ctx wDest
+        verifyAddrs 20 0 addrs
+        let destination = (addrs !! 0) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{2 * minUTxOValue},
```

Comment:
maybe it is good to send 10*minUTxOValue as we risk the situation when sending back from wDest that we can have out+fee+change where change is less than minUTxOValue and hence it is not visible in listAddresses. outputs and changes are required to be > minUTxOValue!

Reply:
Done